app/controllers/navigation/Navigator.scala

```diff
@@ -286,37 +288,41 @@ object Navigator {
   }
 
   val standardCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
-    case CarrierDetails => carrierDetailsPreviousPage
-    case Document       => previousDocumentsPreviousPageDefault
+    case CarrierDetails      => carrierDetailsPreviousPage
+    case Document            => previousDocumentsPreviousPageDefault
+    case RepresentativeAgent => representativeAgentPreviousPage
   }
 
   val standardCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = Map.empty
 
   val supplementaryCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
-    case ConsigneeDetails => consigneeDetailsSupplementaryPreviousPage
-    case Document         => previousDocumentsPreviousPageDefault
+    case ConsigneeDetails    => consigneeDetailsSupplementaryPreviousPage
+    case Document            => previousDocumentsPreviousPageDefault
+    case RepresentativeAgent => representativeAgentPreviousPage
   }
 
   val supplementaryCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = Map.empty
 
   val simplifiedCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
-    case CarrierDetails  => carrierDetailsPreviousPage
-    case Document        => previousDocumentsPreviousPage
-    case DocumentSummary => previousDocumentsSummaryPreviousPage
+    case CarrierDetails      => carrierDetailsPreviousPage
+    case Document            => previousDocumentsPreviousPage
+    case DocumentSummary     => previousDocumentsSummaryPreviousPage
+    case RepresentativeAgent => representativeAgentPreviousPage
   }
 
   val simplifiedCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = Map.empty
 
   val occasionalCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
-    case CarrierDetails  => carrierDetailsPreviousPage
-    case Document        => previousDocumentsPreviousPage
-    case DocumentSummary => previousDocumentsSummaryPreviousPage
+    case CarrierDetails      => carrierDetailsPreviousPage
+    case Document            => previousDocumentsPreviousPage
+    case DocumentSummary     => previousDocumentsSummaryPreviousPage
+    case RepresentativeAgent => representativeAgentPreviousPage
   }
 
   val occasionalCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = Map.empty
 
   val clearanceCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
-    case ExporterDetails     => exporterDetailsClearancePreviousPage
+    case ExporterEoriNumber  => exporterDetailsClearancePreviousPage
```

Comment:
Could you rename this method to something more like `exporterEoriNumberClearancePreviousPage` ?

Reply:
Done